src/framework/components/collision/system.js

```diff
@@ -400,6 +430,19 @@ Object.assign(pc, function () {
             }
 
             pc.CollisionSystemImpl.prototype.updateTransform.call(this, component, position, rotation, scale);
+        },
+
+        remove: function(entity, data) {
+            if (data.shape) {
+                var numShapes = data.shape.getNumChildShapes();
+                for(var i = 0; i < numShapes; i++) {
+                    var shape = data.shape.getChildShape(i);
+                    delete Ammo.btBvhTriangleMeshShape.__cache__[shape.ptr];
```

Comment:
Can you add code comments explaining the requirement for the above line please?

Reply:
I might explore slightly other option, is to cast btShape object into btBvhTriangleMeshShape, and then call Ammo.destroy on it.
Destroying compound shape, does not destroys its children.

Reply:
👍

Reply:
Removed this bit.